airflow/traces/utils.py

```diff
@@ -18,52 +18,66 @@
 from __future__ import annotations
 
 import logging
+from typing import TYPE_CHECKING
 
 from airflow.utils.hashlib_wrapper import md5
 
+if TYPE_CHECKING:
+    from airflow.models import DagRun, TaskInstance
+    from airflow.models.taskinstancekey import TaskInstanceKey
+
 log = logging.getLogger(__name__)
 
 
```

Comment:
I feel like all these `gen_foo` methods can be consolidated.... I'll put some thought into how.  I think maybe something like this..... maybe??

((untested code, just thinking here))
```
def gen_id(seeds: List[str], as_int: bool = False, shorten: bool = False) -> str | int:
    seed_str = "_".join(seeds).encode("utf-8")
    hash_hex = md5(seed_str).hexdigest()
    
    if shorten:
        hash_hex = hash_hex[16:]
    
    return int(hash_hex, 16) if as_int else hash_hex    
    
    
def gen_trace_id(dag_run: DagRun, as_int: bool = False) -> str | int:
    return gen_id([dag_run.dag_id, dag_run.run_id, dag_run.start_date.timestamp], as_int, shorten=False)
    
def gen_span_id_from_ti_key(ti_key: TaskInstanceKey, as_int: bool = False) -> str | int:
    return gen_id([ti_key.dag_id, ti_key.run_id, ti_key.task_id, ti_key.try_number], as_int)     

def gen_dag_span_id(dag_run: DagRun, as_int: bool = False) -> str | int:
    return gen_id([dag_run.dag_id, dag_run.run_id,dag_run.start_date.timestamp()], as_int)
    
<and so on>    
```

or maybe if that is too "magic" then the gen_id method could accept the assembled seed string, so in use it would look like:

```
def gen_trace_id(dag_run: DagRun, as_int: bool = False) -> str | int:
    seed = f"{dag_run.dag_id}_{dag_run.run_id}_{dag_run.start_date.timestamp()}"
    return gen_id(seed, as_int, shorten=False)
```

Either of those would reduce around 50 lines of repetitive code with around 20 lines, and be just as easy to read (IMHO), easier to modify later if all of those needs to get changed at some point, and just as easy to change if only one needs to get adjusted later.  No disadvantage IMHO.

Reply:
I'll give it a few thoughts!

Reply:
I guess by feeding the seed into the gen_id function, that could be done - we could even have `size` instead of `shorten` so I could simply feed in 16 as an input.

Reply:
Yup, I was initially thinking take an int "size" but then thought if there are only two options (16 or 32) then that sounds like a bool.  Either way would work. :+1:

Reply:
If you want to get complicated, you could make a constant which holds the acceptable values for size and only accept those values, but I think that's overkill here.